src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/Repositories/PlayerRepository.cs

```diff
@@ -31,34 +31,27 @@ public PlayerRepository(IUnitOfWork unitOfWork)
             _dalPlayers = _unitOfWork.Context.Players;
         }
 
-        /// <summary>
-        /// Gets unit of work.
-        /// </summary>
-        public IUnitOfWork UnitOfWork
-        {
-            get
-            {
-                return _unitOfWork;
-            }
-        }
-
         /// <summary>
         /// Adds new player.
         /// </summary>
         /// <param name="newEntity">The player for adding.</param>
-        public void Add(Player newEntity)
+        public Player Add(string firstName, string lastName, short? birthYear, short? height, short? weight, int? teamId)
         {
-            var newPlayer = new PlayerEntity();
-            DomainToDal.Map(newPlayer, newEntity);
+            var newPlayer = new Player(firstName, lastName, birthYear, height, weight, teamId);
+            var newEntity = new PlayerEntity();
+
+            DomainToDal.Map(newEntity, newPlayer);
 
-            if (!_dbStorageSpecification.IsSatisfiedBy(newPlayer))
+            if (!_dbStorageSpecification.IsSatisfiedBy(newEntity))
             {
                 throw new InvalidEntityException();
             }
 
-            _dalPlayers.Add(newPlayer);
+            _dalPlayers.Add(newEntity);
             _unitOfWork.Commit();
-            newEntity.Id = newPlayer.Id;
+            newPlayer.Id = newEntity.Id;
+
```

Comment:
![image](https://user-images.githubusercontent.com/11724508/38024026-7b68dd40-328c-11e8-9d0d-cb5bccf79875.png)

Use in:
![image](https://user-images.githubusercontent.com/11724508/38024064-945248c8-328c-11e8-985e-398532004fb1.png)

in tests and playerBuilder

Reply:
use ctor